app.rb

```diff
@@ -3,27 +3,51 @@
 require 'yaml'
 require 'berbix'
 require 'namey'
+require 'json'
+require 'sinatra/custom_logger'
+require 'logger'
+
+set :logger, Logger.new(STDOUT)
 
 berbix_config = YAML.load(File.read('berbix_config.yaml'))
 
 get '/' do
   @generator = Namey::Generator.new
   @username = @generator.name(:common)
-  @uid = rand(10000000000)
+  
+  if params[:uid]
+    @uid = params[:uid]
+  else
+    @uid = 123
+  end
 
   client = Berbix::Client.new(
     client_secret: berbix_config['client_secret'],
     environment: :production
   )
 
-  @transaction_tokens = client.create_transaction(
-    customer_uid: @uid, # ID for the user in client database
-    template_key: berbix_config['template_key'], # Template key for this transaction
-  )
-
-  # To avoid dealing with a DB for this demo, we'll write the refresh token to a file
-  File.open("refresh_token.txt", "w") do |file|
-    file.write(@transaction_tokens.refresh_token)
+  # Check if refresh token exists for given UID
+  if File.file?("#{@uid}_refresh_token.txt")
+    # read the refresh token from disk
+    @refresh_token = File.open("#{@uid}_refresh_token.txt").read
```

Comment:
`/path?uid=../../../etc/passwd`. There's got to be a way to get `File.open` to ignore the second part of a string...